core/src/main/scala/io/qbeast/core/model/RevisionClasses.scala

```diff
@@ -0,0 +1,249 @@
+package io.qbeast.core.model
+
+import com.fasterxml.jackson.annotation.{JsonCreator, JsonValue}
+import com.fasterxml.jackson.databind.annotation.JsonSerialize
+import com.fasterxml.jackson.databind.annotation.JsonSerialize.Typing
+import io.qbeast.IISeq
+import io.qbeast.core.transform.{Transformation, Transformer}
+
+object QTableID {
+
+  @JsonCreator
+  def apply(id: String): QTableID = new QTableID(id)
+
+}
+
+final class QTableID(_id: String) extends Serializable {
+
+  def id: String = _id
+
+  @JsonValue
+  override def toString: String = _id
+
+  def canEqual(other: Any): Boolean = other.isInstanceOf[QTableID]
+
+  override def equals(other: Any): Boolean = other match {
+    case that: QTableID =>
+      toString == that.toString
+    case _ => false
+  }
+
+  override def hashCode(): Int = toString.hashCode
+}
+
+object Revision {
+
+  def firstRevision(
+      tableID: QTableID,
+      desiredCubeSize: Int,
+      columnTransformers: IISeq[Transformer]): Revision = {
+    Revision(
+      0,
+      System.currentTimeMillis(),
+      tableID,
+      desiredCubeSize,
+      columnTransformers,
+      Vector.empty)
+
+  }
+
+}
+
+/**
+ * A revision of a QTable.
+ * @param revisionID the identifier of the revision
+ * @param timestamp the timestamp
+ * @param tableID the table identifier
+ * @param desiredCubeSize the desired cube size
+ * @param columnTransformers the column transformers
+ * @param transformations the space transformations
+ */
+final case class Revision(
+    revisionID: RevisionID,
+    timestamp: Long,
+    tableID: QTableID,
+    desiredCubeSize: Int,
+    @JsonSerialize(
+      as = classOf[IISeq[Transformer]],
+      typing = Typing.STATIC) columnTransformers: IISeq[Transformer],
```

Comment:
Is it a result of scalafmt application?

Reply:
I don't think so, but I have no idea. Maybe @cugni knows better!